File: problem-of-the-day/day02/367 Valid Perfectsquare.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        long long low = 1, high = num;
        long long mid, ans = -1;
        while(low<=high){
            mid = low + (high-low)/2;
            if(mid*mid <=num){
                ans = mid;
                low = mid + 1;
            }
            else{
                high = mid - 1;
            }
        }
        return (ans*ans==num);
    }
};

File: problem-of-the-day/day02/367 Valid Perfectsquare.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        low=1
        high=num
        while(low<=high):
            mid=(low+high)//2
            if mid*mid==num:
                return True
            elif mid*mid>num:
                high=mid-1
            else :
                low=mid+1
        return False